src/transformers/generation_beam_search.py

```diff
@@ -172,13 +172,15 @@ def __init__(
 
         self._is_init = False
         self._beam_hyps = [
-            BeamHypotheses(
-                num_beams=self.num_beams,
-                length_penalty=self.length_penalty,
-                early_stopping=self.do_early_stopping,
-            )
+            [
+                BeamHypotheses(
+                    num_beams=self.num_beams,
```

Comment:
If we are creating `BeamHypotheses` per group for every batch, we might have to send `self.group_size` as the value for `num_beams` parameter.